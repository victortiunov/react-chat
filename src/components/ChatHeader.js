import React from 'react';
import { withStyles } from 'material-ui/styles';
import AppBar from 'material-ui/AppBar';
import Toolbar from 'material-ui/Toolbar';
import Typography from 'material-ui/Typography';
import Avatar from './Avatar';
import ChatMenu from './ChatMenu';
import UserMenu from './UserMenu';

const styles = theme => ({
	appBar: {
		position: 'fixed',
		width: `calc(100% - 320px)`,
	},
	appBarTitle: {
		flex: 1,
		marginLeft: theme.spacing.unit * 2,
		marginRight: theme.spacing.unit * 2,
		color: theme.palette.secondary.contrastText,
	}
});

class ChatHeader extends React.Component {
	render() {
		const { classes, user, editUser, onLogout, activeChat, leaveChat, deleteChat, isConnected } = this.props;

		return (
			< AppBar color="primary" className={classes.appBar} >
				<Toolbar>
					{activeChat ? (
						<React.Fragment>
							<Avatar colorFrom={activeChat._id}>{activeChat.title}</Avatar>
							<Typography variant="title" className={classes.appBarTitle}>
								{activeChat.title}
								<ChatMenu
									user={user}
									onLeaveClick={() => leaveChat(activeChat._id)}
									onDeleteClick={() => deleteChat(activeChat._id)}
									disabled={!isConnected}
								/>
							</Typography>
						</React.Fragment>
					) : (
						<Typography variant="title" className={classes.appBarTitle}>
							DogeCodes React Chat
						</Typography>
					)}
					<UserMenu user={user} editUser={editUser} onLogout={onLogout} disabled={!isConnected} />
				</Toolbar>
			</AppBar >
		)
	}
}

export default withStyles(styles)(ChatHeader);
