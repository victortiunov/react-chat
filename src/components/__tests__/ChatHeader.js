/* eslint-env jest */
import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import ChatHeader from '../ChatHeader';

const mockProps = {
  user: {
    isMember: true,
    isCreator: true,
    isChatMember: true,
  },
  activeChat: {
    _id: '373737',
    title: 'Gryffindor Talks',
  },
  editUser: jest.fn(),
  onLogout: jest.fn(),
  leaveChat: jest.fn(),
  deleteChat: jest.fn(),
  joinChat: jest.fn(),
  isConnected: true,
};

describe('<ChatHeader />', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<ChatHeader {...mockProps} />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders correctly', () => {
    const tree = renderer.create(<ChatHeader {...mockProps} />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders without active chat', () => {
    const tree = renderer.create(<ChatHeader {...mockProps} activeChat={null} />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
