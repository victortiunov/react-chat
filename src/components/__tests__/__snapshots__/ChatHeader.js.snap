// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ChatHeader /> renders correctly 1`] = `
<header
  className="MuiPaper-root-105 MuiPaper-elevation4-111 MuiAppBar-root-97 MuiAppBar-positionFixed-98 MuiAppBar-colorPrimary-103 mui-fixed ChatHeader-appBar-95"
>
  <div
    className="MuiToolbar-root-132 MuiToolbar-gutters-133"
  >
    <div
      className="MuiAvatar-root-134 MuiAvatar-colorDefault-135"
      style={
        Object {
          "backgroundColor": "#8bc34a",
        }
      }
    >
      GT
    </div>
    <h2
      className="MuiTypography-root-137 MuiTypography-title-143 ChatHeader-appBarTitle-96"
    >
      Gryffindor Talks
      <button
        aria-haspopup="true"
        aria-owns={null}
        className="MuiButtonBase-root-167 MuiIconButton-root-161 MuiIconButton-colorInherit-162"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex="0"
        type="button"
      >
        <span
          className="MuiIconButton-label-166"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root-170"
            color={undefined}
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 8c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm0 2c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm0 6c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"
            />
          </svg>
        </span>
        <span
          className="MuiTouchRipple-root-176"
        />
      </button>
    </h2>
    
  </div>
</header>
`;

exports[`<ChatHeader /> renders without active chat 1`] = `
<header
  className="MuiPaper-root-105 MuiPaper-elevation4-111 MuiAppBar-root-97 MuiAppBar-positionFixed-98 MuiAppBar-colorPrimary-103 mui-fixed ChatHeader-appBar-95"
>
  <div
    className="MuiToolbar-root-132 MuiToolbar-gutters-133"
  >
    <h2
      className="MuiTypography-root-137 MuiTypography-title-143 ChatHeader-appBarTitle-96"
    >
      DogeCodes React Chat
    </h2>
    
  </div>
</header>
`;
